src/Property/Selector.php

```diff
@@ -74,7 +75,9 @@ public function getSelector(): string
 
     public function setSelector(string $selector): void
     {
-        $this->selector = \trim($selector);
+        $selector = \trim($selector);
+
+        $this->selector = !str_contains($selector, '[') ? preg_replace('/\s{2,}/', ' ', $selector) : $selector;
```

Comment:
The `\s` needs double-escaping (it does technically work single-escaped, but fails a coding style check).

We could also convert any whitespace to an actual space - e.g. a single newline, tab, etc. - and have a test for that too.

And for better performance (and to avoid [catastrophic backtracking](https://www.regular-expressions.info/catastrophic.html) in the unlikely case of a selector string with a million spaces being passed), we can use the [possessive quantifier](https://www.regular-expressions.info/possessive.html) (`+`).


```suggestion
        $this->selector = !str_contains($selector, '[') ? preg_replace('/\\s++/', ' ', $selector) : $selector;
```